### PR TITLE
feat(prometheus-server): develop TOSIT role for prometheus-server

### LINKED ISSUE #27: Use prometheus-community/ansible instead of cloudalchemy/ansible-prometheus after deprecation
[Cloud Alchemy Prometheus Role](https://github.com/cloudalchemy/ansible-prometheus) has recently been deprecated in favor of [Prometheus Community Ansible role](https://github.com/prometheus-community/ansible).

### PR DESCRIPTION
**Pull request by giem-git**
#### Which issue(s) this PR fixes

Fixes #27 and #28 

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: roles/prometheus/server/tasks/install.yml [hunk 1]
```diff
@@ -2,9 +2,39 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ---
-- ansible.builtin.include_role:
-    name: cloudalchemy.prometheus
-    tasks_from: preflight
+- name: Ensure prometheus group exists
+  group:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/install.yml:6]: ```ansible.builtin.group```

File: roles/prometheus/server/tasks/install.yml [hunk 2]
```diff
@@ -2,9 +2,39 @@
...
+- name: Create data directory
+  file:
+    path: "{{ prometheus_db_dir }}"
+    state: directory
+    owner: "{{ prometheus_user }}"
+    group: "{{ prometheus_group }}"
+    mode: "755"
+
+- name: Create configuration directories
+  file:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/install.yml:28]: ```ansible.builtin.file```

File: roles/prometheus/server/tasks/install.yml [hunk 3]
```diff
@@ -51,6 +78,12 @@
     - prometheus
     - promtool
 
-- ansible.builtin.include_role:
-    name: cloudalchemy.prometheus
-    tasks_from: install
+- name: Template Prometheus service file
+  template:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/install.yml:82]: ```ansible.builtin.template```

File: roles/prometheus/server/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+- name: systemctl daemon-reload
+  systemd:
```
Comment (kpgtek) [inline roles/prometheus/server/handlers/main.yml:3]: With Ansible Core 2.15, I think it is better to use absolute path in Ansible modules' names:
```ansible.builtin.systemd```

File: roles/prometheus/server/tasks/config.yml [hunk 5]
```diff
@@ -57,10 +57,43 @@
   when: item.value.group in groups.keys()
   loop: "{{ prometheus_tdp_targets | dict2items }}"
 
-- ansible.builtin.include_role:
-    name: cloudalchemy.prometheus
-    tasks_from: preflight
+- name: Template prometheus.yml
+  template:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/config.yml:61]: ```ansible.builtin.template```

File: roles/prometheus/server/tasks/config.yml [hunk 6]
```diff
@@ -57,10 +57,43 @@
...
+    src: prometheus.yml.j2
+    dest: "{{ prometheus_config_dir }}/prometheus.yml"
+    owner: root
+    group: "{{ prometheus_group }}"
+    mode: "640"
+    validate: "{{ prometheus_binary_install_dir }}/promtool check config %s"
+
+- name: Configure Prometheus web
+  copy:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/config.yml:70]: ```ansible.builtin.copy```

File: roles/prometheus/server/tasks/config.yml [hunk 7]
```diff
@@ -57,10 +57,43 @@
...
+    content: "{{ prometheus_web_config | to_nice_yaml(indent=2) }}"
+    dest: "{{ prometheus_config_dir }}/web.yml"
+    owner: root
+    group: "{{ prometheus_group }}"
+    mode: "640"
+
+- name: Alerting rules file
+  template:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/config.yml:78]: ```ansible.builtin.template```

File: roles/prometheus/server/tasks/config.yml [hunk 8]
```diff
@@ -57,10 +57,43 @@
...
+    src: "alert.rules.j2"
+    dest: "{{ prometheus_config_dir }}/rules/ansible_managed.rules"
+    owner: root
+    group: "{{ prometheus_group }}"
+    mode: "640"
+    validate: "{{ prometheus_binary_install_dir }}/promtool check rules %s"
+  when: prometheus_alert_rules != []
+
+- name: Configure service discovery targets
+  copy:
```
Comment (kpgtek) [inline roles/prometheus/server/tasks/config.yml:88]: ```ansible.builtin.copy```

**Comment by kpgtek** [comment]
Do we need a ```systemctl reload prometheus``` Ansible task?
I think ```/usr/lib/systemd/system/prometheus.service``` provides this.

**Comment by giem-git** [comment]
> Do we need a `systemctl reload prometheus` Ansible task?
> I think `/usr/lib/systemd/system/prometheus.service` provides this.

When the unit file or source configuration of a service is changed on disk, `systemctl daemon-reload` must be run to reload units.

**Comment by kpgtek** [comment]
> > Do we need a `systemctl reload prometheus` Ansible task?
> > I think `/usr/lib/systemd/system/prometheus.service` provides this.
> 
> When the unit file or source configuration of a service is changed on disk, `systemctl daemon-reload` must be run to reload units.

As we discussed and the same as with other components, ```systemcdt reload``` task is not used.

**Review by kpgtek — APPROVED** [review]

File: roles/prometheus/server/tasks/config.yml [hunk 9]
```diff
@@ -52,15 +60,11 @@
                  {'files': [prometheus_static_targets_dir + '/' + item.value.component + '.yml']}
                  ],
                'relabel_configs': prometheus_relabels
-           }] 
-      }}
+           }]
+      }}#
```
[the remark this answers is not in the document]
Comment (giem-git) [reply roles/prometheus/server/tasks/config.yml]: not needed. fixed

File: roles/prometheus/server/tasks/config_set_web_config.yml [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+  ansible.builtin.set_fact:
+    old_hashed_password: ''
+  when: not web_cfg.stat.exists
+
+- name: "File exists, try to read old password"
+  when: web_cfg.stat.exists
+  block:
+    - name: Read prometheus web config file
+      ansible.builtin.command:
+        cmd: "cat {{ prometheus_config_dir }}/web.yml"
```
[the remark this answers is not in the document]
Comment (giem-git) [reply roles/prometheus/server/tasks/config_set_web_config.yml]: done

File: roles/prometheus/server/tasks/config_set_web_config.yml [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    - name: Get old hash password from web config
+      ansible.builtin.set_fact:
+        old_hashed_password: "{{ (web_yml.stdout | from_yaml).basic_auth_users[prometheus_admin_user] | default('') }}"
+
+- name: Generate new hashed password if we don't have one
+  ansible.builtin.set_fact:
+    prometheus_admin_hashed_password: "{{ prometheus_admin_password | password_hash('blowfish') }}"
+  when: old_hashed_password == ''
+
+- name: Compute hashed password based on old salt
```
[the remark this answers is not in the document]
Comment (giem-git) [reply roles/prometheus/server/tasks/config_set_web_config.yml]: done

**Review by rpignolet — APPROVED** [review]

**Pull request merged**